lib/_http_client.js

```diff
@@ -24,6 +24,18 @@ function ClientRequest(options, cb) {
     if (!options.hostname) {
       throw new Error('Unable to determine the domain name');
     }
+  } else if (options instanceof url.URL) {
+    options = {
+      protocol: options.protocol,
+      host: options.host,
+      hostname: options.hostname,
+      port: options.port,
+      hash: options.hash,
+      search: options.search,
+      pathname: options.pathname,
+      path: `${options.pathname}${options.search}`,
+      href: options.href
+    };
```

Comment:
I'd rather avoid doing the for-in loop in favor of keeping things more explicit and obvious. Also, as you point out, the `auth` vs `username`+`password` would demonstrates that the properties do not match up one-to-one.